src/dbnode/storage/bootstrap/bootstrapper/README.md

```diff
@@ -2,12 +2,74 @@
 
 The collection of bootstrappers comprise the task executed when bootstrapping a node.
 
+## A high level overview of bootstrapping process
+
+During the bootstrapping process, the list of configured bootstrappers is iterated, invoking the bootstrap routine of each of them. Each bootstrapper is given time ranges for which data needs to be bootstrapped. It acknowledges, retrieves, processes and/or persists the data that is available to it, and returns a subset of requested time ranges that it was unable to fulfill. 
+
+If there are any errors or unfulfilled time ranges after going through all the configured bootstrappers, the process is retried. 
+
+See pseudo-code below for more details.
+
+```python
+bootstrappers = {filesystem, commitlog, peer, uninitialized}
+
+while true:
+    namespaces = getOwnedNamespaces()
+    shards = filterNonBootstrapped(getOwnedShards())
```

Comment:
Needs a callout here for peers bootstrapper i think - it takes INITIALIZING shards.

Reply:
https://github.com/m3db/m3/pull/3203/commits/241e69f468515145c05fed2537891dad9bc29517